### DISCUSSION TITLE
How to add strokes to vector lines

### DISCUSSION THREAD

**Discussion by CodeK1988** [mapsforge/vtm#1106] [Q&A] [answered]
How to add strokes to vector lines.

Style.builder().buffer(0.5)
                /*.strokeColor(polylineOptions.getLineColor())
                .strokeWidth(polylineOptions.getLineWidth())*/
                .fillColor(Color.parseColor("#E03232"))
                .stippleWidth(6f)
                .strokeColor(Color.parseColor("#ffFAFAFA"))
                .strokeWidth(1f)
                .build();

I want to distinguish the fill color and stroke color. How can I do it?

**Comment by devemux86** [comment, marked as answer]
See how the roads are rendered in the map themes with 2 lines:
- Above a thin line
- Below a thick line

**Reply by CodeK1988** [reply]
.strokeColor(Color.parseColor("#E03232"))
.strokeWidth(6f)
.fillColor(Color.WHITE)

not working,  @devemux86 
hi mate, Can you be more detailed? Or do I have to draw two lines?


<img width="227" alt="1709026261998" src="https://github.com/mapsforge/vtm/assets/13393518/6328bde4-8946-42a6-96e2-3abb35435836">

**Reply by devemux86** [reply]
> Or do I have to draw two lines?

2 lines: 1 thick below + 1 thin above.

**Reply by CodeK1988** [reply]
@devemux86 Do you mean the red line covers the white line?

**Reply by devemux86** [reply]
If want red fill with white stroke, then red line over white line.
